sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2318,16 +2364,17 @@ pi_result cuda_piQueueRelease(pi_queue command_queue) {
 }
 
 pi_result cuda_piQueueFinish(pi_queue command_queue) {
-
-  // set default result to a negative result (avoid false-positve tests)
-  pi_result result = PI_OUT_OF_HOST_MEMORY;
+  pi_result result = PI_SUCCESS;
 
   try {
 
     assert(command_queue !=
            nullptr); // need PI_ERROR_INVALID_EXTERNAL_HANDLE error code
     ScopedContext active(command_queue->get_context());
-    result = PI_CHECK_ERROR(cuStreamSynchronize(command_queue->stream_));
+
+    command_queue->for_each_stream([&result](CUstream s) mutable {
```

Comment:
Does this need to be `mutable`?

Reply:
Yes, otherwise result is captured by const reference.

Reply:
Are you sure? To my knowledge capture by reference isn't `const` unless you force it to be or the captured variable is `const`.

Reply:
it seems you are right.